nitric/api.go

```diff
@@ -219,30 +216,27 @@ func (m *manager) newApi(name string, opts ...ApiOption) (Api, error) {
 		}
 	}
 
+	registerChan := make(chan RegisterResult)
+
 	// declare resource
-	_, err = rsc.Declare(context.TODO(), &resourcev1.ResourceDeclareRequest{
+	go defaultManager.registerResource(&resourcev1.ResourceDeclareRequest{
```

Comment:
If it's important that register is always called in a go routine and all the callers need to create their own channel, could the model be flipped so the register method starts the go routine internally and returns a 'receive only' channel?

Reply:
Basically so it's harder to use it wrong